### ISSUE TITLE
Incorrect function parameter type reference in certain cases

### ISSUE THREAD

**Issue by dimarudnev** [TypeStrong/typedoc#1124]
- [x] I have checked [issues with bug label](https://github.com/TypeStrong/typedoc/labels/bug) and found no duplicates

Steps to reproduce:
1. Copy the attached test.d.ts (from [typedoc-reference.zip](https://github.com/TypeStrong/typedoc/files/3792521/typedoc-reference.zip)) to a new folder
2. Run
```
typedoc --json res.json --ignoreCompilerErrors --includeDeclarations --excludePrivate --target ES5 .
```
3. Open res.json
4. Find the setMasterFilter method (id=157025)
5. Find the "values" parameter (id=157027) of the setMasterFilter method
6. See the "values" parameter type id.
![image](https://user-images.githubusercontent.com/13471814/67926331-d2c91280-fbc6-11e9-92eb-59070ccd819a.png)

Current Behavior: id is **157021** that refers to the **MasterFilterValues property** of the ItemState class
![image](https://user-images.githubusercontent.com/13471814/67926419-073cce80-fbc7-11e9-9f33-f0cf00d4b4b5.png)

Expected Behavior: id is **157022** and refers to the **MasterFilterValues Type** alias.
![image](https://user-images.githubusercontent.com/13471814/67926440-158aea80-fbc7-11e9-8e38-ae406b8ff08e.png)

## Environment
  - Typedoc version: 0.15.0
  - Node.js version: v10.16.3
  - OS: Windows 10

**Comment by Gerrit0** [comment]
This happens because the TypeScript compiler interns (some) identical types for performance reasons. https://github.com/microsoft/TypeScript/issues/28197

Fixing this might prove... difficult. PR welcome if you'd like to give it a shot. I probably won't get to it for at least a couple of months.

**Comment by Gerrit0** [comment]
I have no idea why I thought this was a problem with interning.... this is a complex enough type that it shouldn't be happening here. Interning only really happens with `type Alias = <some primitive>`. 

This specific case appears to have been fixed by 0.20 (currently `typedoc@beta`). Hovering over the link shows that it targets the correct reflection. This also revealed a bug with 0.20's converter, it was producing multiple identical reflections for the namespace, so thanks for that!

![image](https://user-images.githubusercontent.com/19329837/100491890-92e3a000-30e4-11eb-81c7-1924e47d3687.png)
